### ISSUE TITLE
MSBuildWorkspace.OpenSolutionAsync throws ArgumentException

### ISSUE THREAD

**Issue by JosefPihrt** [dotnet/roslyn#30382]
**Version Used**: 2.9.0

**Steps to Reproduce**:

1. Clone https://github.com/dotnet/BenchmarkDotNet

1. Create new console app and run

```csharp
VisualStudioInstance instance = MSBuildLocator.QueryVisualStudioInstances().First();

MSBuildLocator.RegisterInstance(instance);

using (MSBuildWorkspace workspace = MSBuildWorkspace.Create())
{
    Solution solution = await workspace.OpenSolutionAsync(solutionFilePath);
}
```
```
System.ArgumentException: Item with Same Key has already been added.

   v System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   v System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   v System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   v Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.<LoadProjectInfosFromPathAsync>d__20.MoveNext()
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.<TryLoadAndAddReferenceAsync>d__33.MoveNext()
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.<ResolveReferencesAsync>d__32.MoveNext()
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.<>c__DisplayClass21_0.<<CreateProjectInfoAsync>b__0>d.MoveNext()
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.<DoOperationAndReportProgressAsync>d__17`1.MoveNext()
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.<LoadProjectInfosFromPathAsync>d__20.MoveNext()
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.<LoadAsync>d__18.MoveNext()
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.<LoadSolutionInfoAsync>d__22.MoveNext()
--- Konec trasování zásobníku z předchozího místa, ze kterého byla vyvolána výjimka ---
   v System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   v System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   v Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.<OpenSolutionAsync>d__23.MoveNext()
```

Build passes in Visual Studio.

I observed similar behavior with https://github.com/Fody/PropertyChanged.

**Comment by DustinCampbell** [comment]
Thanks for the report! I can reproduce this in both projects mentioned.

**Comment by DustinCampbell** [comment]
I have a fix for this issue in #30490
